### ISSUE TITLE
Using leptos_actix::redirect in a server function with JS disabled in Firefox or Chromium results in an empty response

### ISSUE THREAD

**Issue by jim-taylor-business** [leptos-rs/leptos#2506]
Using `leptos_actix::redirect` in a server function with JS disabled in Firefox or Chromium results in an empty response.

**Leptos Dependencies**

From standard `cargo leptos new` template

```rust
use leptos::*;
use leptos_router::*;

#[server(LoginFn, "/serverfn")]
pub async fn login() -> Result<(), ServerFnError> {
    use leptos_actix::redirect;
    redirect("/?some=thing");
    Ok(())
}

#[component]
pub fn App() -> impl IntoView {
    let login = create_server_action::<LoginFn>();
    view! {
        <ActionForm action=login>
            <button type="submit">
                "Login"
            </button>
        </ActionForm>
    }
}
```

**To Reproduce**
1. Disable JS in Firefox or Chromium
2. Go to `/`
3. Click on Login
4. See blank screen

**Expected behavior**
Browser should be redirected to `/?some=thing`.

**Screenshots**
![image](https://github.com/leptos-rs/leptos/assets/7439891/149b6c31-3ab1-44c7-afb0-c24f324e3b3b)

**Additional context**
This has worked in most previous versions of Leptos.

**Comment by gbj** [comment]
Thanks! See the comment in #2507 for an explanation. I tested Axum and it seems like this bug doesn't exist in that integration.

**Comment by jim-taylor-business** [comment]
> (This has probably been broken since 0.6.0)

yep, although i thought it got fixed around 0.6.7 :stuck_out_tongue_winking_eye:
